pkg/controller/agent/vlanconfig/controller.go

```diff
@@ -166,12 +166,15 @@ func (h Handler) setupVLAN(vc *networkv1.VlanConfig) error {
 		goto updateStatus
 	}
 	// set up VLAN
-	v = vlan.NewVlan(vc.Spec.ClusterNetwork, localAreas)
-	setupErr = v.Setup(uplink)
+	v = vlan.NewVlan(vc.Spec.ClusterNetwork)
+	if setupErr = v.Setup(uplink); setupErr != nil {
+		goto updateStatus
```

Comment:
a `goto` method is preferred when the normal method does not apply to you, in here I would suggest us to refactor to call a normal method in the future version.

Reply:
OK